src/frame-writer.cpp

```diff
@@ -124,12 +136,40 @@ AVPixelFormat FrameWriter::get_input_format()
     case INPUT_FORMAT_X2BGR10:
         return AV_PIX_FMT_X2BGR10LE;
 #endif
+    case INPUT_FORMAT_DMABUF:
+        return AV_PIX_FMT_VAAPI;
     default:
         std::cerr << "Unknown format: " << params.format << std::endl;
         std::exit(-1);
     }
 }
 
+static const struct {
+    int drm;
+    AVPixelFormat av;
+} drm_av_format_table [] = {
+    { GBM_FORMAT_NV12, AV_PIX_FMT_NV12 },
+    { GBM_FORMAT_ARGB8888, AV_PIX_FMT_BGRA },
```

Comment:
Why are we mapping RGB to BGR and vice versa? It's done elsewhere in wf-recorder too but I don't really understand it

Reply:
little/big endian